src/Datadog.Trace/SpanContextPropagator.cs

```diff
@@ -166,6 +164,37 @@ public SpanContext Extract<T>(T carrier, Func<T, string, IEnumerable<string>> ge
             }
         }
 
+        private static TraceId ParseTraceId<T>(T headers, string headerName)
+            where T : IHeadersCollection
+        {
+            var headerValues = headers.GetValues(headerName);
+            return ParseTraceId(headerValues, headerName);
+        }
+
+        private static TraceId ParseTraceId<T>(T carrier, Func<T, string, IEnumerable<string>> getter, string headerName)
+        {
+            var headerValues = getter(carrier, headerName).ToList();
+            return ParseTraceId(headerValues, headerName);
+        }
+
+        private static TraceId ParseTraceId(IEnumerable<string> headerValues, string headerName)
+        {
+            var headerValuesList = headerValues.ToList();
```

Comment:
Same `ToList()` applies here

Reply:
Yep, the IEnumerable is enumerated twice in this method, so I did that.

Reply:
`headerValues` is only enumerated twice if there is a trace id header with an invalid value. The different is probably very small, but it seems like this is trading performance in the common case vs the exceptional case.